traffic_portal/app/src/common/modules/form/deliveryService/form.deliveryService.DNS.tpl.html

```diff
@@ -941,7 +941,7 @@ <h3 ng-if="!open()">Previous Value</h3>
                         </div>
                         </label>
                         <div class="col-md-10 col-sm-10 col-xs-12">
-                            <input id="geoLimitCountries" name="geoLimitCountries" type="text" class="form-control" ng-model="deliveryService.geoLimitCountries" maxlength="255" pattern="[A-Z]+(,[A-Z]+)*">
+                            <input id="geoLimitCountries" name="geoLimitCountries" type="text" class="form-control" ng-model="deliveryService.geoLimitCountries" maxlength="255" pattern="(\[[A-Z]+(,[A-Z]+)*\]|[A-Z]+(,[A-Z]+)*)">
```

Comment:
Users should not be writing loose JSON syntax into our inputs, they should be able to handle that for the user. This document fragment actually doesn't need to change at all to make this change to Delivery Services, though IMO it would be an improvement to make it a true variable list of inputs like the "Consistent Hash Query Parameters".

Reply:
Alright, so I'm going to back out the TP changes, but on the API side, I'll add the logic for the `geoLimitCountries` to be able to be either a string or an array of strings. Does that sound good?

Reply:
yeah, that should be fine.